### PR TITLE
clean up consensus scp crate integration tests

### PR DESCRIPTION
**Pull request by rjwalters**
Cargo expects integration tests to live in a `tests` directory at the same level as `src` or  `bench`. All rust files in the `tests` directory are compiled and run as integration tests. Common code should go in subfolders following the `mod.rs` naming convention.

see https://doc.rust-lang.org/book/ch11-03-test-organization.html

This PR refactors the scp crate tests to follow cargo conventions without other changes, in preparation for later adding additional integration tests against new network topologies.

### CODE REVIEW

File: consensus/scp/tests/mock_network/mod.rs [hunk 1]
```diff
@@ -1,21 +1,21 @@
 // Copyright (c) 2018-2020 MobileCoin Inc.
-#![allow(unused_attributes)]
-//TODO -- which attribute is unused???
+
+// The separated integration tests do not use all common code.
+// TODO: consider breaking out new_mesh and new_cyclic into separate integration code source.
+#![allow(dead_code)]
```
Comment (eranrund) [inline consensus/scp/tests/mock_network/mod.rs:5]: Whats being unused?

File: consensus/scp/tests/test_mesh_networks.rs [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    // See https://stackoverflow.com/questions/39204908/how-to-check-release-debug-builds-using-cfg-in-rust/39205417#39205417
+    if cfg!(debug_assertions) {
+        return;
+    }
+
+    mesh_test_helper(9, 7, logger);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// Other network topologies
```
Comment (eranrund) [inline consensus/scp/tests/test_mesh_networks.rs:163]: Why are these here? They're not mesh. If you want to break this into separate files (which honestly I don't personally find particularly useful, but I don't feel strongly about) then this shouldn't be here.

Comment (rjwalters) [reply consensus/scp/tests/test_mesh_networks.rs:163]: The fig2 network from the paper doesn't actually have quorom intersection, which is why this test is marked as ignored. I would be happy to just delete it if there are no objections.

File: consensus/scp/tests/mock_network/mod.rs (hunk 1 above)
Comment (rjwalters) [reply consensus/scp/tests/mock_network/mod.rs:5]: the mesh tests don't use `new_cyclic()` and the cyclic tests don't use `new_mesh()`... I have a fix ready to go but wanted to keep this PR as small as possible.

File: consensus/scp/tests/test_mesh_networks.rs (hunk 2 above)
Comment (rjwalters) [reply consensus/scp/tests/test_mesh_networks.rs]: (I deleted it.)
